site/app/models/gradeable/GradeableUtils.php

```diff
@@ -91,4 +92,19 @@ public static function getAllGradeableListFromUserId(Core $core, User $user): ar
         $core->getConfig()->setCourseLoaded(false);
         return ["gradeables" => $gradeables, "graded_gradeables" => $graded_gradeables, "submit_btns" => $submit_btns];
     }
+
+    /**
+     * This function assumes that you are calling it where Core has a defined course
+     *
+     * @return \app\models\Button[][]|Gradeable[][]|GradedGradeable[][]|array[]
+     * @throws \Exception
+     */
+    public static function getGradeablesFromUserAndCourse(Core $core, array &$calendar_messages): array {
+        $gradeables_of_course = self::getGradeablesFromCourse($core, $core->getConfig()->getSemester(), $core->getConfig()->getCourse(), $calendar_messages, false);
+        $gradeables = $gradeables_of_course["gradeables"];
+        $graded_gradeables = $gradeables_of_course["graded_gradeables"];
+        $submit_btns = $gradeables_of_course["submit_btns"];
+
+        return ["gradeables" => $gradeables, "graded_gradeables" => $graded_gradeables, "submit_btns" => $submit_btns];
```

Comment:
This line is pretty long. It might be better to wrap it for readability purposes.